src/coreclr/jit/codegenloongarch64.cpp

```diff
@@ -1924,7 +1924,33 @@ void CodeGen::genCodeForNegNot(GenTree* tree)
 //
 void CodeGen::genCodeForBswap(GenTree* tree)
 {
-    NYI_LOONGARCH64("genCodeForBswap unimpleement yet");
+    assert(tree->OperIs(GT_BSWAP, GT_BSWAP16));
+
+    emitAttr  attr      = emitActualTypeSize(tree);
+    regNumber targetReg = tree->GetRegNum();
+    emitter*  emit      = GetEmitter();
+
+    GenTree* operand = tree->gtGetOp1();
+    assert(!operand->isContained());
+    // The src must be a register.
+    regNumber   operandReg = genConsumeReg(operand);
+    instruction ins;
+
+    if (tree->OperIs(GT_BSWAP16))
+    {
+        ins = INS_revb_4h;
```

Comment:
Does this zero the upper 16 bits? `BSWAP16` is defined to zero the upper bits.

Reply:
No, it swaps the bytes for the upper 16 bits too. Actually it swaps the whole 64bit register in 4x 2-byte groups.

> zero the upper bits

Is it `genCanOmitNormalizationForBswap16`? A zero extend would be required if normalization can't be omitted?

Reply:
> Is it `genCanOmitNormalizationForBswap16`? A zero extend would be required if normalization can't be omitted?

Right